calc.py

```diff
@@ -0,0 +1,53 @@
+import math
+
+
+def add(a, b):
+    check_types(a, b)
+    return a + b
+
+
+def subtract(a, b):
+    check_types(a, b)
+    return a - b
+
+
+def multiply(a, b):
+    check_types(a, b)
+    return a * b
+
+
+def log(a):
+    check_type(a)
+    return math.log(a)
+
+
+def log10(a):
+    check_type(a)
+    return math.log10(a)
+
+
+def check_types(a, b):
+    check_type(a, 'First')
+    check_type(b, 'Second')
+
+
+def check_type(param, name=''):
+    if param is None:
+        raise TypeError
```

Comment:
If the parameters are None, instead of returning None, I have changed it to raise TypeError and have made corresponding change in the test_add.py i.e. to check whether TypeError is raised when one or both the input parameters to add function are None.